model/riscv_insts_zcmp.sail

```diff
@@ -0,0 +1,643 @@
+    0x9  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xa  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xb  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b00000, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xc  => reg_list = [0b00000, 0b00000, 0b00000, 0b00000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xd  => reg_list = [0b00000, 0b00000, 0b00000, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xe  => reg_list = [0b00000, 0b00000, 0b11001, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+    0xf  => reg_list = [0b11011, 0b11010, 0b11001, 0b11000, 0b10111, 0b10110, 0b10101, 0b10100, 0b10011, 0b10010, 0b01001, 0b01000, 0b00001],
+  };
+
+
+  var stack_adj_base : bits(xlen) = zeros();
+  var stack_adj : bits(xlen) = zeros();
+
+  if (sizeof(xlen) == 32) then {
+    match rlist {
+      0x4 => {stack_adj_base = zero_extend(0x10);},
+      0x5 => {stack_adj_base = zero_extend(0x10);}, 
+      0x6 => {stack_adj_base = zero_extend(0x10);},
+      0x7 => {stack_adj_base = zero_extend(0x10);},
+      0x8 => {stack_adj_base = zero_extend(0x20);},
+      0x9 => {stack_adj_base = zero_extend(0x20);},
+      0xa => {stack_adj_base = zero_extend(0x20);},
+      0xb => {stack_adj_base = zero_extend(0x20);}, 
+      0xc => {stack_adj_base = zero_extend(0x30);}, 
+      0xd => {stack_adj_base = zero_extend(0x30);}, 
+      0xe => {stack_adj_base = zero_extend(0x30);}, 
+      0xf => {stack_adj_base = zero_extend(0x40);},
+    };
+    } else {
+    match rlist {
+      0x4 => {stack_adj_base = zero_extend(0x10);},
+      0x5 => {stack_adj_base = zero_extend(0x10);},
+      0x6 => {stack_adj_base = zero_extend(0x20);},
+      0x7 => {stack_adj_base = zero_extend(0x20);},
+      0x8 => {stack_adj_base = zero_extend(0x30);},
+      0x9 => {stack_adj_base = zero_extend(0x30);},
+      0xa => {stack_adj_base = zero_extend(0x40);},
+      0xb => {stack_adj_base = zero_extend(0x40);}, 
+      0xc => {stack_adj_base = zero_extend(0x50);}, 
+      0xd => {stack_adj_base = zero_extend(0x50);}, 
+      0xe => {stack_adj_base = zero_extend(0x60);}, 
+      0xf => {stack_adj_base = zero_extend(0x70);},
+    };
+  };
+
+  var stack_adj_by : bits(xlen) = zero_extend(sp54 << 4);
+  stack_adj = stack_adj_base + stack_adj_by;
+  
+  var addr : bits(xlen) = X(sp) - zero_extend(bytes);
+
+  foreach (i from 0 to 12) {
+    //if register i is in xreg_list
+    if (reg_list[i] != 0b00000) then {
+      if (bytes == 0x4) then {
+        let rs1 = reg_list[i];
+        let imm : bits(xlen) = stack_adj - zero_extend(bytes);
+        let _ = execute(STORE(imm[11..0], rs1, sp, WORD, false, false))
```

Comment:
And if one of these doesn't retire successfully you just steamroll ahead ignoring the error entirely...